status/manager.go

```diff
@@ -0,0 +1,174 @@
+package status
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/lightninglabs/lightning-terminal/litrpc"
+)
+
+// SubServerStatus represents the status of a sub-server.
+type SubServerStatus struct {
+	// Disabled is true if the sub-server is available in the LiT bundle but
+	// has explicitly been disabled by the user.
+	Disabled bool
+
+	// Running is true if the sub-server is enabled and has successfully
+	// been started.
+	Running bool
+
+	// Err will be a non-empty string if the sub-server failed to start.
+	Err string
+}
+
+// newSubServerStatus constructs a new SubServerStatus.
+func newSubServerStatus() *SubServerStatus {
+	return &SubServerStatus{
+		Disabled: true,
+	}
+}
+
+// Manager managers the status of any sub-server registered to it. It is also an
+// implementation of the litrpc.StatusServer which can be queried for the status
+// of various LiT sub-servers.
+type Manager struct {
```

Comment:
I'm probably missing something, but why can't the `subservers.Manager` keep track of disabled/running/error and we just have a simple RPC server struct that gets an instance of said manager?

Reply:
the main reason is cause `subservers.Manager` is just meant to handle the external daemons and does not handle Lit and LND (both of which are kind of in their own category) .So seemed a bit contrived to try squash those in to the `subservers.Manager` since they dont fit the `SubServer` interface.

Reply:
Ah, okay, I didn't consider that. Okay, makes sense to keep them separate then.

Reply:
lemme know what you think :)